### PR TITLE
feat(core): not allowing to send empty PIN in T2B1

### PR DESCRIPTION
**Pull request by grdddj**
Making it impossible to send empty PIN from the PIN keyboard on `T2B1`

Also added click tests for ignoring empty PIN for both models

### CODE REVIEW

File: core/embed/rust/src/ui/model_tr/component/input_methods/pin.rs [hunk 1]
```diff
@@ -262,7 +262,14 @@ where
                     self.show_real_pin = true;
                     self.update(ctx);
                 }
-                PinAction::Enter => return Some(CancelConfirmMsg::Confirmed),
+                PinAction::Enter => {
```
Comment (matejcik) [inline core/embed/rust/src/ui/model_tr/component/input_methods/pin.rs:265]:
```suggestion
                PinAction::Enter if !self.is_empty() => {
```

Comment (grdddj) [reply core/embed/rust/src/ui/model_tr/component/input_methods/pin.rs]: Good point, done in the force-push

**Review by mmilata — APPROVED** [review]

**Comment by bosomt** [comment]
**QA OK**

**Info:**
 - Suite version: desktop 23.10.1 (6c8d061f7305947b0270a2560e4cc65d1f7ded27)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuite/23.10.1 Chrome/112.0.5615.204 Electron/24.5.1 Safari/537.36
 - OS: MacIntel
 - Screen: 1440x900
 - Device: Trezor T2B1 2.6.2 regular (revision 6e5967af25de87bebd331d697e6ef940e64286ef)
 - Transport: BridgeTransport 2.0.33
